packages/mobile/src/components/drawers/OfflineListeningDrawer.tsx

```diff
@@ -0,0 +1,217 @@
+import { requestDownloadAllFavorites } from 'app/store/offline-downloads/slice'
+import { makeStyles } from 'app/styles'
+import { useThemeColors } from 'app/utils/theme'
+
+import { NativeDrawer } from '../drawer'
+
+const useDrawerStyles = makeStyles(({ spacing, palette, typography }) => ({
+  container: {
+    paddingTop: spacing(4),
+    paddingBottom: spacing(10),
+    flexDirection: 'column',
+    paddingHorizontal: spacing(4),
+    rowGap: spacing(6),
+    alignItems: 'center'
+  },
+  title: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingTop: spacing(2)
+  },
+  titleText: {
+    textTransform: 'uppercase',
+    marginTop: spacing(4)
+  },
+  descriptionText: {
+    textAlign: 'center',
+    lineHeight: typography.fontSize.large * 1.3
+  },
+  descriptionContainer: {
+    width: '100%'
+  },
+  titleIcon: {
+    position: 'relative',
+    top: 7,
+    color: palette.neutral,
+    marginRight: spacing(3)
+  }
+}))
+
+const useToggleStyles = makeStyles(({ spacing, palette }) => ({
+  toggleContainer: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    width: '100%',
+    alignItems: 'center'
+  },
+  titleContainer: {
+    columnGap: spacing(2),
+    flexDirection: 'row'
+  }
+}))
+
+const messages = {
+  offlineListeningTitle: 'Offline Listening',
+  offlineListeningDescription:
+    'Use the toggles to select what you’d like synced for offline streaming.',
+  comingSoonToggleSuffix: '(coming soon...)',
+  favorites: 'Favorites',
+  reposts: 'Reposts',
+  purchased: 'Purchased',
+  saveChanges: 'Save Changes'
+}
+
+const OfflineListeningOptionToggle = ({
+  title,
+  icon: Icon,
+  value,
+  onValueChange,
+  disabled
+}: {
+  title: string
+  icon: ComponentType<SvgProps>
+  value: boolean
+  onValueChange?: (value: boolean) => void | Promise<void>
+  disabled?: boolean
+}) => {
+  const styles = useToggleStyles()
+  const { neutral, neutralLight4 } = useThemeColors()
+
+  return (
+    <View style={styles.toggleContainer}>
+      <View style={styles.titleContainer}>
+        <Icon
+          fill={disabled ? neutralLight4 : neutral}
+          height={20}
+          width={20}
+        />
+        <Text
+          weight='demiBold'
+          fontSize='large'
+          color={disabled ? 'neutralLight4' : 'neutral'}
+        >
+          {title}
+        </Text>
+      </View>
+      <Switch value={value} disabled={disabled} onValueChange={onValueChange} />
+    </View>
+  )
+}
+
+export const OfflineListeningDrawer = () => {
+  const styles = useDrawerStyles()
+  const { neutralLight2 } = useThemeColors()
+  const dispatch = useDispatch()
+  const { data, onClose } = useDrawer('OfflineListening')
+  const { isFavoritesMarkedForDownload, onSaveChanges } = data
+
+  const [isFavoritesOn, setIsFavoritesOn] = useState(
+    isFavoritesMarkedForDownload
+  )
+
+  const handleSaveChanges = useCallback(() => {
+    if (isFavoritesMarkedForDownload) {
```

Comment:
could maybe collapse the if else's into something a bit more explicit:
`if (isFavoritesMarkedForDownload && !isFavoritesOn) {...} else if (!isFavoritesMarkedForDownload && isFavoritesOn) {...} ? main idea being to make clear we do stuff when the value has changed